test/dogma/rules/comparison_to_boolean_test.exs

```diff
@@ -0,0 +1,59 @@
+defmodule Dogma.Rules.ComparisonToBooleanTest do
+  @comp [:==, :===, :!=, :!==]
+  @bools [true, false]
+
+  use DogmaTest.Helper
+
+  alias Dogma.Rules.ComparisonToBoolean
+  alias Dogma.Script
+  alias Dogma.Error
+
+  def test(script) do
+    script
+    |> Script.parse("foo.ex")
+    |> ComparisonToBoolean.test
+  end
+
+  def errors(count) do
+    for line_no <- (count..1) do
+      %Error{
+        rule: ComparisonToBoolean,
+        message: "Comparison to a boolean is usually pontless",
+        position: line_no
+      }
+    end
+  end
+
+  with "lefthand booleans" do
+    setup context do
+      errors = for fun <- @comp, bool <- @bools, into: "" do
+        "foo #{fun} #{bool}\n"
+      end |> test
+      %{errors: errors}
+    end
+
+    errors(8) |> should_register_errors
+  end
+
+  with "righthand booleans" do
+    setup context do
+      errors = for fun <- @comp, bool <- @bools, into: "" do
+        "#{bool} #{fun} foo\n"
+      end |> test
+      %{errors: errors}
+    end
+
+    errors(8) |> should_register_errors
+  end
+
+  with "vars on both sides" do
+    setup context do
+      errors = for fun <- @comp, into: "" do
+        "foo #{fun} bar\n"
+      end |> test
+      %{errors: errors}
+    end
+
```

Comment:
There is so much metaprogramming in this set of tests that it's not possible to tell what we're testing, or what the expected behaviour is.

Tests should be simple, declarative, and easy to understand. Can we re-write this file please? Verbosity is fine.